tuna/miopen/celery_tuning/tuning.py

```diff
@@ -424,3 +398,51 @@ def drain_queue(worker_type, dbt, event):
         continue
 
   return True
+
+async def consume(job_counter, worker_type, dbt):
+  """Retrieve celery results from redis db"""
+  redis = await aioredis.from_url(
+      f"redis://{TUNA_CELERY_BROKER}:{TUNA_REDIS_PORT}/15")
+  LOGGER.info('Connected to redis')
+  while job_counter.value > 0:
+    cursor = "0"
+    keys = []
+    while cursor != 0:
+      cursor, results = await redis.scan(cursor, match='*'
+                                        )  # update with the celery pattern
+      keys.extend(results)
+    LOGGER.info('Found results')
+    for key in keys:
+      LOGGER.info(type(key))
+      try:
+        data = await redis.get(key)
+        if data:
+          await result_callback(key.decode('utf-8'), data.decode('utf-8'), worker_type, dbt)
+          await redis.delete(key)
+          job_counter.value - 1
+      except aioredis.exceptions.ResponseError as red_err:
+        LOGGER.error(red_err)
+        LOGGER.info(key.decode('utf-8'))
+
+    LOGGER.info('Processed results')
+    await asyncio.sleep(1)
+  LOGGER.info('Job counter reached 0')
+  await redis.close()
+
+async def result_callback(key, data, worker_type, dbt):
+  """Function callback for celery async jobs to store results"""
+  data = json.loads(data)
+  LOGGER.info(data)
+
+  with DbSession() as session:
+    fin_json = data['result']['ret']
+    context = data['result']['context']
+    LOGGER.info('Parsing: %s', fin_json)
+    LOGGER.info('Parsing context: %s', context)
+    if worker_type == 'fin_build_worker':
+      process_fin_builder_results(session, fin_json, context, dbt)
+    elif worker_type == 'fin_eval_worker':
+      process_fin_evaluator_results(session, fin_json, context, dbt)
+    else:
+      raise ValueError('Unsupported worker_type')
+
```

Comment:
Is this still required?

Reply:
if you mean the callback function, then yes still required, it does the whole parsing of results.